GameDataParser/Parsers/MapParser.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Xml;
+using GameDataParser.Crypto.Common;
+using GameDataParser.Files;
+using Maple2Storage.Types;
+using Maple2Storage.Types.Metadata;
+
+namespace GameDataParser.Parsers
+{
+    public class MapParser : Exporter<List<MapMetadata>>
+    {
+        public MapParser(MetadataResources resources) : base(resources, "map") { }
+
+        protected override List<MapMetadata> Parse()
+        {
+            // Iterate over preset Cubes to later reference while iterating over exported maps
+            List<string> mapCubes = new List<string>();
+            foreach (PackFileEntry entry in Resources.ExportedFiles)
+            {
+                if (!entry.Name.StartsWith("flat/presets/presets cube/"))
+                {
+                    continue;
+                }
+
+                // Check if file is valid
+                string objStr = entry.Name.ToLower();
+                if (string.IsNullOrEmpty(objStr))
+                {
+                    continue;
+                }
+                if (mapCubes.Contains(objStr))
+                {
+                    continue;
+                }
+
+                // Parse XML
+                XmlDocument document = Resources.ExportedMemFile.GetDocument(entry.FileHeader);
+                XmlElement root = document.DocumentElement;
+                string cubeName = root.Attributes["name"].Value.ToLower();
+                mapCubes.Add(cubeName);
+            }
+
+            // Parse every block for each map
+            List<MapMetadata> mapsList = new List<MapMetadata>();
+            foreach (PackFileEntry entry in Resources.ExportedFiles.Where(x => x.Name.StartsWith("xblock/")))
+            {
+                if (entry.Name.Contains("_cn.xblock") || entry.Name.Contains("_jp.xblock") || entry.Name.Contains("_kr.xblock"))
+                {
+                    continue;
+                }
+
+                string mapIdStr = Regex.Match(entry.Name, @"\d{8}").Value;
+                if (string.IsNullOrEmpty(mapIdStr))
+                {
+                    continue;
+                }
+
+                MapMetadata metadata = new MapMetadata();
+                int mapId = int.Parse(mapIdStr);
+                metadata.Id = mapId;
```

Comment:
Might as well remove the `mapId` variable since you don't use it anywhere else and just set `metadata.Id` directly

Reply:
sure